src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -33,8 +40,16 @@ namespace AppInstaller::CLI::Workflow
         bool IsInstallerBetterMatch(
             const Manifest::ManifestInstaller& installer1,
             const Manifest::ManifestInstaller& installer2,
-            Manifest::ManifestInstaller::InstallerTypeEnum installedType)
+            Manifest::InstallerTypeEnum installedType)
         {
+            bool isOSVersionSatisfied1 = installer1.MinOSVersion.empty() || Runtime::IsCurrentOSVersionGreaterThanOrEqual(Utility::Version(installer1.MinOSVersion));
+            bool isOSVersionSatisfied2 = installer2.MinOSVersion.empty() || Runtime::IsCurrentOSVersionGreaterThanOrEqual(Utility::Version(installer2.MinOSVersion));
+
+            if (isOSVersionSatisfied1 && !isOSVersionSatisfied2)
+            {
+                return true;
+            }

```

Comment:
Let's re-write the loop to check `IsInstallerApplicable` first before ever calling `IsInstallerBetterMatch`.  That way we don't need to duplicate these kinds of boolean checks.  It ends up making this function just the arch > arch comparison.  Not sure why I didn't think of it when I rewrote this a while back.